From 22aa77fc17d3d8aae4308b46ceefaa8a151c9dee Mon Sep 17 00:00:00 2001
From: Gergely Nagy <algernon@balabit.hu>
Date: Fri, 6 Dec 2013 15:04:57 +0100
Subject: [PATCH] logproto: Fix the framed server free method

Upstream-Status: Backport

The LogProtoFramedServer free method did not call the free_method of its
super class, which resulted in memory & FD leaks when using a syslog()
source. This patch corrects that.

Reported-by: Andras Mitzki <micek@balabit.hu>
Signed-off-by: Gergely Nagy <algernon@balabit.hu>
---
 lib/logproto/logproto-framed-server.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/logproto-framed-server.c b/lib/logproto-framed-server.c
index 7a547f3..5c7d268 100644
--- a/lib/logproto-framed-server.c
+++ b/lib/logproto-framed-server.c
@@ -271,6 +271,8 @@ log_proto_framed_server_free(LogProtoSer
 {
   LogProtoFramedServer *self = (LogProtoFramedServer *) s;
   g_free(self->buffer);
+
+  log_proto_server_free_method(s);
 }
 
 LogProtoServer *
-- 
1.8.4.1

